### ISSUE TITLE
Tags limit was increased in Azure to 50, but azurerm provider still have 15

### ISSUE THREAD

**Issue by Shenderchuk** [terraform-providers/terraform-provider-azurerm#4070]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform v0.11.13
+ provider.azurerm v1.32.1
The same for (Terraform v0.12.5 + provider.azurerm v1.32.1)

### Affected Resource(s)

All `azurerm_XXXXX` resources with tags.

### Terraform Configuration Files

```hcl
provider "azurerm" {
  version = "~> 1.32.1"
}

resource "azurerm_resource_group" "resource_group" {
  name     = "tagstesting"
  location = "westeurope"

  tags {
    "01" = "01"
    "02" = "02"
    "03" = "03"
    "04" = "04"
    "05" = "05"
    "06" = "06"
    "07" = "07"
    "08" = "08"
    "09" = "09"
    "10" = "10"
    "11" = "11"
    "12" = "12"
    "13" = "13"
    "14" = "14"
    "15" = "15"
    "16" = "16"
  }
}
```

### Expected Behavior

Terraform should deploy a resource group with 16 tags in this case, but in general it should be able to deploy any Azure resource with up to 50 tags (except Storage Account, there is still limited with 15 tags ). Please see new Azure documentation: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-using-tags

Also, it is possible to create more than 15 tags via Azure portal.

### Actual Behavior

Error: azurerm_resource_group.resource_group: a maximum of 15 tags can be applied to each ARM resource

### Steps to Reproduce

1. `terraform apply`

**Comment by Shenderchuk** [comment]
Added PR which fixes it
https://github.com/terraform-providers/terraform-provider-azurerm/pull/4071

**Comment by tombuildsstuff** [comment]
Fixed via #4071 - thanks @Shenderchuk
